### PR TITLE
Epmlstrtai 704

### PR DESCRIPTION
**Pull request by OrlovaDasha**
1) Create ProjectCreateDTO and ProjectUpdateDTO + changed Project[Full]DTO
2) Move ProjectDTOs, Project controller to controllers.Project package
3) Remove Dozer ProjectTransformer and create custom ProjectTransformer and tests for it

### CODE REVIEW

File: src/main/java/com/epam/test_generator/controllers/Project/ProjectController.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
-package com.epam.test_generator.controllers;
+package com.epam.test_generator.controllers.Project;
```
Comment (pomkine) [inline src/main/java/com/epam/test_generator/controllers/Project/ProjectController.java]: Let's avoid using upper case in package names

File: src/main/java/com/epam/test_generator/controllers/Project/request/ProjectCreateDTO.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+public class ProjectCreateDTO {
+
+    @NotNull
+    @Size(min = 1, max = 255)
+    private String name;
+
+    @Size(max = 255)
+    private String description;
+
+    private String jiraKey;
```
Comment (pomkine) [inline src/main/java/com/epam/test_generator/controllers/Project/request/ProjectCreateDTO.java]: Do we expecting jiraKey from UI?

File: src/test/java/com/epam/test_generator/dto/ProjectTransformerTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    private static final Long ID = 1L;
+    private static final String NAME = "NAME";
+    private static final String DESCRIPTION = "DESCRIPTION";
+    private static final String JIRA_KEY = "KEY";
+
+    private Suit suit;
+    private SuitDTO suitDTO;
+
+    @Before
+    public void setUp() throws Exception {
```
Comment (aleksei-khitev) [inline src/test/java/com/epam/test_generator/dto/ProjectTransformerTest.java]: Do we really need to add exception declaration here?

File: src/test/java/com/epam/test_generator/dto/ProjectTransformerTest.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    public void updateFromDto_ProjectUpdateDTO_Project_Success() {
+        when(suitTransformer.fromDto(any(SuitDTO.class))).thenReturn(suit);
+
+        Project expectedProject = new Project();
+        expectedProject.setName(NAME);
+        expectedProject.setDescription(DESCRIPTION);
+        expectedProject.setJiraKey(JIRA_KEY);
+        expectedProject.setActive(true);
+        expectedProject.setUsers(Collections.emptySet());
+        expectedProject.setSuits(Collections.singletonList(suit));
```
Comment (aleksei-khitev) [inline src/test/java/com/epam/test_generator/dto/ProjectTransformerTest.java]: You have similar declaration of a project more then 2 times. It looks like copy/paste.
May be you should move duplication to the setUp() method?

File: src/main/java/com/epam/test_generator/controllers/project/request/ProjectUpdateDTO.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    @Size(min = 1, max = 255)
+    private String name;
+
+    @Size(max = 255)
+    private String description;
+
+    @Size(max = 255)
+    private String jiraKey;
+
+    private Boolean active;
```
Comment (pomkine) [inline src/main/java/com/epam/test_generator/controllers/project/request/ProjectUpdateDTO.java]: I'm not 100% sure but now we should not have an ability to deactivate project.

**Pull request merged**